core/src/test/java/io/questdb/test/griffin/engine/functions/json/JsonPathFunctionFactoryIntTest.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    @Test
+    public void testDifferentType() throws SqlException {
+        callFn(utf8("{\"path\": \"abc\"}"), utf8(".path")).andAssert(Integer.MIN_VALUE);
+        callFn(dirUtf8("{\"path\": \"abc\"}"), dirUtf8(".path")).andAssert(Integer.MIN_VALUE);
+    }
+
+    @Test
+    public void testEmptyJson() throws SqlException {
+        callFn(utf8("{}"), utf8(".path")).andAssert(Integer.MIN_VALUE);
+    }
+
+    @Test
+    public void testFloat() throws SqlException {
+        callFn(dirUtf8("{\"path\": 123.45}"), dirUtf8(".path")).andAssert(Integer.MIN_VALUE);
+    }
+
+    @Test
+    public void testInString() throws SqlException {
+        callFn(utf8("{\"path\": \"123\"}"), utf8(".path")).andAssert(123);
+    }
+
+    @Test
+    public void testNegative() throws SqlException {
+        callFn(utf8("{\"path\": -123}"), utf8(".path")).andAssert(-123);
+    }
+
+    @Test
+    public void testNullJson() throws SqlException {
+        callFn(
+                utf8(null),
+                utf8(".path")
+        ).andAssert(Integer.MIN_VALUE);
+        callFn(
+                utf8(null),
+                dirUtf8(".path")
+        ).andAssert(Integer.MIN_VALUE);
+    }
+
+    @Test
+    public void testNullJsonValue() throws SqlException {
+        callFn(utf8("{\"path\": null}"), utf8(".path")).andAssert(Integer.MIN_VALUE);
+        callFn(utf8("{\"path\": null}"), dirUtf8(".path")).andAssert(Integer.MIN_VALUE);
+        callFn(dirUtf8("{\"path\": null}"), utf8(".path")).andAssert(Integer.MIN_VALUE);
+        callFn(dirUtf8("{\"path\": null}"), dirUtf8(".path")).andAssert(Integer.MIN_VALUE);
+    }
+
+    @Test
+    public void testOne() throws SqlException {
+        callFn(utf8("{\"path\": 1}"), utf8(".path")).andAssert(1);
+    }
+
+    @Test
+    public void testStrict10000() throws SqlException {
+        callFn(utf8("{\"path\": 10000}"), utf8(".path"), FAIL_ON_ERROR).andAssert(10000);
+    }
+
+    @Test
+    public void testStrictBigNumber() throws SqlException {
+        final CairoException exc = Assert.assertThrows(
+                CairoException.class,
+                () -> callFn(utf8("{\"path\": 100000000000000000000000000}"), utf8(".path"), FAIL_ON_ERROR)
+        );
+        Assert.assertTrue(exc.getMessage().contains("json_path(.., '.path'): INCORRECT_TYPE:"));
+    }
+
+    @Test
+    public void testStrictDefaultDifferentType() throws SqlException {
+        final CairoException exc = Assert.assertThrows(
+                CairoException.class,
+                () -> callFn(utf8("{\"path\": \"abc\"}"), utf8(".path"), FAIL_ON_ERROR, 42));
+        Assert.assertTrue(exc.getMessage().contains("json_path(.., '.path'): INCORRECT_TYPE:"));
+    }
+
+    @Test
+    public void testStrictDefaultNullJsonValue() throws SqlException {
+        callFn(utf8("{\"path\": null}"), utf8(".path"), FAIL_ON_ERROR, 42).andAssert(Integer.MIN_VALUE);
+    }
+
+    @Test
+    public void testStrictDifferentType() throws SqlException {
```

Comment:
unused exception (throughout the class)

Reply:
Yup, and not just this class. As mentioned.. code's not ready yet. Full of these little things. Also formatting is off, missing features etc.